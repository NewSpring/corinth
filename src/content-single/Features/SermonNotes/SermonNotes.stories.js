import React from 'react';
import { storiesOf } from '@apollosproject/ui-storybook';

import SermonNotes from '.';

const features = [
  {
    props: {
      children: [
        {
          props: {
            id: 'TextFeature:123',
            card: false,
            header: false,
            body: 'This is a text feature',
            contentId: 'WeekendContentItem:123',
            sharing: { message: 'This is a text feature' },
          },
        },
      ],
    },
  },
];

storiesOf('features/SermonNotes', module)
  .add('default', () => <SermonNotes />)
  .add('one speaker', () => (
    <SermonNotes
      title={'A Fun Sunday'}
      series={'A Fun Series'}
      communicators={[
        { firstName: 'Bradford', nickName: 'Brad', lastName: 'Cooper' },
      ]}
      features={features}
    />
  ));
