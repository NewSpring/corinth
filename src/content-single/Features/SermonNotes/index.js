import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import {
  H3,
  PaddedView,
  ActionCard,
  Touchable,
  Icon,
  H5,
  FlexedView,
  styled,
  BodyText,
} from '@apollosproject/ui-kit';
import { LegalText } from '@apollosproject/ui-scripture';
import share from '../../../utils/content/share';

const ExportWrapper = styled({
  flexDirection: 'row',
})(FlexedView);

const PaddedText = styled(({ theme }) => ({
  paddingHorizontal: theme.sizing.baseUnit,
}))(BodyText);

const SermonNotes = ({ contentItem, features }) => {
  const {
    communicators = [],
    guestCommunicators = [],
    title = '',
    seriesConnection = { series: { title: '' }, itemIndex: 0 },
  } = contentItem;
  const [sharedMsg, changeSharedMsg] = useState('');
  const [enhancedFeatures, enhanceFeatures] = useState([]);
  const [copyrights, setCopyrights] = useState(new Set());
  const onNotesChange = (id, text) => {
    const placeholder = `${id}{{(.*?)}}`;
    const re = new RegExp(placeholder, 'gs');
    changeSharedMsg((msg) => msg.replace(re, `${id}{{${text}}}`));
  };
  const communicatorNames = communicators.map(
    ({ nickName, firstName, lastName }) =>
      `${nickName || firstName} ${lastName}`
  );
  const speakers = communicatorNames.concat(guestCommunicators);

  // assemble exported notes
  useEffect(() => {
    // add title, series, and speakers to top
    let msg = `${title}\n${seriesConnection.series.title}\n`;
    speakers.forEach((speaker) => {
      msg += `${speaker}\n`;
    });
    msg += '\n';

    // loop through all features and add them
    const featuresWithCallbacks = features.map((feature) => {
      const featureProps = feature.props.children[0].props;
      let cleanedFeature = feature;

      // clean scripture features
      if (featureProps.scriptures) {
        // pop out all copyrights to be used later
        featureProps.scriptures.forEach((s) => {
          copyrights.add(s.copyright);
        });
        setCopyrights(copyrights);

        // remove all copyrights from features
        const cleanedScriptures = featureProps.scriptures.map((s) => ({
          ...s,
          copyright: '',
        }));
        cleanedFeature = {
          ...feature,
          props: {
            ...feature.props,
            children: [
              {
                ...feature.props.children[0],
                props: {
                  ...feature.props.children[0].props,
                  scriptures: cleanedScriptures,
                },
              },
              feature.props.children[1],
            ],
          },
        };
      }

      // assemble starting message without custom notes
      if (featureProps.sharing) {
        msg = `${msg + featureProps.sharing.message}\n\n`;
        return cleanedFeature;
      }

      // drop in placeholders for custom notes
      if (featureProps.id.match(/NoteFeature/g).length > 0)
        msg = `${msg + featureProps.id}{{}}\n\n`;

      // add callbacks to swap note placeholders with custom text
      return {
        ...feature,
        props: {
          ...feature.props,
          children: [
            {
              ...feature.props.children[0],
              props: {
                ...feature.props.children[0].props,
                onNotesChange,
              },
            },
            feature.props.children[1],
          ],
        },
      };
    });
    if (msg !== '') changeSharedMsg(msg);
    enhanceFeatures(featuresWithCallbacks);
  }, []);

  return (
    <ActionCard
      action={
        // TODO: use this when it can accept on a custom onPress function
        // so we can share content item info if we want to
        // https://github.com/ApollosProject/apollos-prototype/issues/1014
        // <ShareContentButtonConnected
        // message={sharedMsg}
        // icon={'play'}
        // itemId={contentId}
        // />
        <Touchable
          onPress={() => {
            console.log(contentItem.id); // left in the prop for the to do item above
            const message = sharedMsg.replace(
              /\w+Feature:\w+{{(.*?)}}\n\n/gs,
              (match, p1) => (p1 === '' ? p1 : `${p1}\n\n`)
            );
            share({ message });
          }}
        >
          <ExportWrapper>
            <PaddedText>Export</PaddedText>
            <Icon name={'export'} size={24} />
          </ExportWrapper>
        </Touchable>
      }
    >
      <H3>Sermon Notes</H3>
      <H5>{title}</H5>
      <H5>
        {seriesConnection.series.title}
        {seriesConnection.itemIndex
          ? ` - Week ${seriesConnection.itemIndex}`
          : ''}
        {/*  - Date */}
      </H5>
      {speakers[0] != null
        ? speakers.map((speaker) =>
            speaker !== '' ? <H5 key={speaker}>{speaker}</H5> : null
          )
        : null}

      <PaddedView />
      {enhancedFeatures}
      {[...copyrights].map((copyright) => (
        <LegalText key={copyright}>{copyright}</LegalText>
      ))}
    </ActionCard>
  );
};

SermonNotes.propTypes = {
  contentItem: PropTypes.shape({
    id: PropTypes.string,
    communicators: PropTypes.arrayOf(
      PropTypes.shape({
        firstName: PropTypes.string,
        lastName: PropTypes.string,
      })
    ),
    guestCommunicators: PropTypes.arrayOf(PropTypes.string),
    title: PropTypes.string,
    seriesConnection: PropTypes.shape({
      series: PropTypes.shape({ title: PropTypes.string }),
      itemIndex: PropTypes.number,
    }),
  }),
  features: PropTypes.arrayOf(PropTypes.element),
};

SermonNotes.defaultProps = {
  features: [],
  contentItem: {},
};

export default SermonNotes;
