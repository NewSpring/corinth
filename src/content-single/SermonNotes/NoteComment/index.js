import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Mutation } from '@apollo/client/react/components';
import GET_SERMON_NOTES from '../getSermonNotes';
import NoteComment from './NoteComment';
import SAVE_NOTES_COMMENT from './saveNotesComment';

const NoteCommentConnected = ({ contentID, noteID, onChange, initialText }) => {
  const [timer, setTimer] = useState(null);
  const [textToSave, setText] = useState(initialText);
  const handleChange = (saveComment) => (newText) => {
    setText(newText);
    onChange(newText);

    // autosave after 3 sec
    // check for existing timeout, clear it, and set a new one
    clearTimeout(timer);
    const newTimer = setTimeout(saveComment, 3000);
    setTimer(newTimer);
  };

  // pre-fill sermon note exports
  useEffect(
    () => {
      onChange(initialText);
    },
    [onChange, initialText]
  );
  return (
    <Mutation
      mutation={SAVE_NOTES_COMMENT}
      variables={{ contentID, noteID, text: textToSave }}
      update={(cache, { data: { saveNotesComment: comment } }) => {
        // get all sermon notes for this content item
        const data = cache.readQuery({
          query: GET_SERMON_NOTES,
          variables: { contentID },
        });

        // replace comment text
        data.node.sermonNotes = data.node.sermonNotes.map(
          (note) => (note.id === noteID ? { ...note, comment } : note)
        );

        // write notes back out
        cache.writeQuery({
          query: GET_SERMON_NOTES,
          variables: { contentID },
          data,
        });
      }}
    >
      {(saveComment) => (
        <NoteComment
          onChange={handleChange(saveComment)}
          initialText={initialText}
        />
      )}
    </Mutation>
  );
};

NoteCommentConnected.propTypes = {
  contentID: PropTypes.string,
  noteID: PropTypes.string,
  onChange: PropTypes.func,
  initialText: PropTypes.string,
};

export default NoteCommentConnected;
