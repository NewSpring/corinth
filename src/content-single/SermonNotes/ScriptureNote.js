import React from 'react';
import PropTypes from 'prop-types';
import { ScriptureItem } from '@apollosproject/ui-scripture';

const ScriptureNote = ({ scripture: { reference, html, version } }) => (
  <ScriptureItem reference={reference} html={html} version={version} />
);

ScriptureNote.propTypes = {
  scripture: PropTypes.shape({
    copyright: PropTypes.string,
    reference: PropTypes.string,
    html: PropTypes.string,
    version: PropTypes.string,
  }),
};

export default ScriptureNote;
