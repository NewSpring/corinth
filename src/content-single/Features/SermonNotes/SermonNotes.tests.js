import React from 'react';
import renderer from 'react-test-renderer';
import Providers from 'newspringchurchapp/src/Providers';

import SermonNotes from '.';

// const features = [
// {
// props: {
// children: [
// {
// props: {
// id: 'TextFeature:123',
// card: false,
// header: false,
// body: 'This is a text feature',
// contentId: 'WeekendContentItem:123',
// sharing: { message: 'This is a text feature' },
// },
// },
// ],
// },
// },
// ];

describe('The SermonNotes component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Providers>
        <SermonNotes />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render with a title, series, and one speaker', () => {
    const tree = renderer.create(
      <Providers>
        <SermonNotes
          contentItem={{
            title: 'A Fun Sunday',
            communicators: [
              { firstName: 'Bradford', nickName: 'Brad', lastName: 'Cooper' },
            ],
            seriesConnection: {
              series: { title: 'A Fun Series' },
              itemIndex: 4,
            },
          }}
          // TODO figure this out, should try and snap some notes too
          // features={features}
        />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
