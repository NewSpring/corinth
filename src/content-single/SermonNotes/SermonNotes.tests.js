import React from 'react';

import { Providers, renderWithApolloData } from '@apollosproject/ui-test-utils';
import getSermonNotes from './getSermonNotes';
import SermonNotes from './SermonNotes';
import SermonNotesConnected from '.';

const contentItemStuff = {
  id: 'WeekendContentItem:123',
  __typename: 'WeekendContentItem',
  title: 'a sermon',
  seriesConnection: {
    __typename: 'SeriesConnection',
    series: { __typename: 'ContentSeriesContentItem', title: 'a series' },
  },
  communicators: [],
  guestCommunicators: [],
};

const notes = [
  {
    id: 'NotesTextBlock:123',
    __typename: 'NotesTextBlock',
    simpleText: '1. Point One',
    allowsComment: false,
    isHeader: true,
    hasBlanks: true,
    hiddenText: '1. Point ___',
    comment: null,
  },
  {
    id: 'NotesTextBlock:234',
    __typename: 'NotesTextBlock',
    simpleText: 'Sub point',
    allowsComment: true,
    isHeader: false,
    hasBlanks: false,
    hiddenText: null,
    comment: {
      id: 'NotesBlockComment:123',
      __typename: 'NotesBlockComment',
      text: 'So good!',
    },
  },
  {
    id: 'NotesScriptureBlock:123',
    __typename: 'NotesScriptureBlock',
    simpleText:
      '11) The greatest among you will be your servant.  Matthew 23:11',
    allowsComment: true,
    scripture: {
      __typename: 'Scripture',
      reference: 'Matthew 23:11',
      html:
        '<p class="p"><span data-number="11" data-sid="MAT 23:11" class="v">11</span><span class="wj">The greatest among you will be your servant.</span> </p>',
      copyright:
        '\n          The Holy Bible, New International Version® NIV®\n          Copyright © 1973, 1978, 1984, 2011 by Biblica, Inc.®\n          Used by Permission of Biblica, Inc.® All rights reserved worldwide.\n        ',
      version: 'NIV',
    },
    comment: null,
  },
];

const emptyMock = {
  request: {
    query: getSermonNotes,
    variables: { contentID: 'WeekendContentItem:123' },
  },
  result: {
    data: {
      node: {
        sermonNotes: [],
        ...contentItemStuff,
      },
    },
  },
};

const notesMock = { sermonNotes: notes, ...contentItemStuff };

describe('the sermon notes', () => {
  it('renders a custom note block when notes are blank', async () => {
    const tree = await renderWithApolloData(
      <Providers mocks={[emptyMock]}>
        <SermonNotesConnected contentID={'WeekendContentItem:123'} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders sermon notes', async () => {
    const tree = await renderWithApolloData(
      <Providers>
        <SermonNotes isLoading={false} {...notesMock} />
      </Providers>
    );
    expect(tree).toMatchSnapshot();
  });
});
