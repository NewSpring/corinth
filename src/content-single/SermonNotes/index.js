import React from 'react';
import PropTypes from 'prop-types';
import { Query } from '@apollo/client/react/components';
import { ErrorCard } from '@apollosproject/ui-kit';
import SermonNotes from './SermonNotes';
import GET_SERMON_NOTES from './getSermonNotes';

const SermonNotesConnected = ({ contentID }) => (
  <Query query={GET_SERMON_NOTES} variables={{ contentID }}>
    {({ data: { node = {} } = { node: {} }, loading, error }) => {
      if (error) return <ErrorCard error={error} />;
      if (loading || !node || !node.sermonNotes?.length) return null;
      return <SermonNotes isLoading={loading} {...node} />;
    }}
  </Query>
);

SermonNotesConnected.propTypes = {
  contentID: PropTypes.string,
};

export default SermonNotesConnected;
