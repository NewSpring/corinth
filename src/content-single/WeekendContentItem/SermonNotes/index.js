import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import SermonNotes from './SermonNotes';
import GET_SERMON_NOTES from './getSermonNotes';

const SermonNotesConnected = ({ contentID }) => (
  <Query query={GET_SERMON_NOTES} variables={{ contentID }}>
    {({ data: { node = {} }, loading }) => (
      <SermonNotes isLoading={loading} {...node} />
    )}
  </Query>
);

SermonNotesConnected.propTypes = {
  contentID: PropTypes.string,
};

export default SermonNotesConnected;
