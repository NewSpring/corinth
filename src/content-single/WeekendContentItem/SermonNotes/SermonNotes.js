import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Share } from 'react-native';
import {
  ActionCard,
  H3,
  H5,
  BodyText,
  FlexedView,
  styled,
  Touchable,
  PaddedView,
  Icon,
} from '@apollosproject/ui-kit';
import { LegalText } from '@apollosproject/ui-scripture';
import { AnalyticsConsumer } from '@apollosproject/ui-analytics';
import TextNote from './TextNote';
import ScriptureNote from './ScriptureNote';
import CustomNote from './CustomNote';

const NoteTypeMapper = (props) => {
  switch (props.type) {
    case 'TextNote':
      return <TextNote {...props} />;
    case 'ScriptureNote':
      return <ScriptureNote {...props} />;
    default:
      return null;
  }
};

NoteTypeMapper.propTypes = {
  type: PropTypes.string,
};

const ExportWrapper = styled({
  flexDirection: 'row',
})(FlexedView);

const SermonNotes = ({ isLoading, ...contentItem }) => {
  const hasNotes = !isLoading && !contentItem.sermonNotes.length;
  const [exports, setExports] = useState([]);
  const handleChangeExportText = (index) => (text) => {
    const changed = exports;
    changed[index] = text;
    setExports(changed);
  };
  const {
    title = '',
    seriesConnection: { series: { title: seriesTitle } } = {
      series: { title },
    },
    communicators = [],
    guestCommunicators = [],
    sermonNotes = [],
  } = contentItem;
  const communicatorNames = communicators.map(
    ({ nickName, firstName, lastName }) =>
      `${nickName || firstName} ${lastName}`
  );
  const speakers = communicatorNames.concat(guestCommunicators);
  const copyrights = new Set(
    sermonNotes
      .filter(({ scripture }) => !!scripture)
      .map((note) => note.scripture.copyright)
  );

  // set up exported text
  useEffect(() => {
    const exportTemplate = [];
    sermonNotes.forEach((note) => {
      exportTemplate.push(note.simpleText);
      exportTemplate.push('');
    });
    setExports(exportTemplate);
  }, []);

  return (
    <ActionCard
      isLoading={isLoading}
      action={
        hasNotes ? (
          <AnalyticsConsumer>
            {({ track }) => (
              <Touchable
                onPress={() => {
                  const textExports = exports.filter((text) => text !== '');
                  Share.share({
                    message: `Sermon Notes\n${title}\n${seriesTitle}${
                      speakers.length ? '\n' : ''
                    }${speakers.join('\n')}${
                      textExports.length ? '\n\n' : ''
                    }${textExports.join('\n\n')}`,
                  });
                  track({
                    eventName: 'Exported Sermon Notes',
                    properties: { title },
                  });
                }}
              >
                <ExportWrapper>
                  <PaddedView vertical={false}>
                    <BodyText>Export</BodyText>
                  </PaddedView>
                  <Icon name={'export'} size={24} />
                </ExportWrapper>
              </Touchable>
            )}
          </AnalyticsConsumer>
        ) : null
      }
    >
      <H3 isLoading={false}>{hasNotes ? 'Sermon Notes' : null}</H3>
      <H5>{title}</H5>
      <H5>{seriesTitle}</H5>
      {speakers.length
        ? speakers
            .filter((speaker) => speaker !== '')
            .map((speaker) => <H5 key={speaker}>{speaker}</H5>)
        : null}

      {sermonNotes.length ? (
        <>
          {sermonNotes.map((note, i) => (
            <>
              <NoteTypeMapper type={note.__typename} key={note.id} {...note} />
              {note.allowsCustomNote ? (
                <CustomNote onChange={handleChangeExportText(i * 2 + 1)} />
              ) : null}
            </>
          ))}
          {[...copyrights].map((text) => (
            <LegalText key={text}>{text}</LegalText>
          ))}
        </>
      ) : null}
    </ActionCard>
  );
};

SermonNotes.propTypes = {
  isLoading: PropTypes.bool,
};

export default SermonNotes;
