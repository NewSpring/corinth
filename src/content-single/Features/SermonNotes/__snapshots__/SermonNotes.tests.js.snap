// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SermonNotes component should render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 16,
      "marginHorizontal": 16,
      "marginVertical": 12,
      "shadowColor": "rgba(8, 0, 20, 0.25)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 6,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 16,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Black",
            "fontSize": 24,
            "lineHeight": 27.6,
          }
        }
      >
        Sermon Notes
      </Text>
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Medium",
            "fontSize": 14,
            "lineHeight": 20.16,
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Medium",
            "fontSize": 14,
            "lineHeight": 20.16,
          }
        }
      >
        
        
      </Text>
      <View
        style={
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-end",
          "paddingHorizontal": 16,
          "paddingTop": 0,
          "paddingVertical": 16,
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            bold={false}
            italic={false}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Regular",
                "fontSize": 16,
                "lineHeight": 26,
                "paddingHorizontal": 16,
              }
            }
          >
            Export
          </Text>
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                undefined,
                null,
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={512}
            vbWidth={576}
            width={24}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGPath
                d="M572.29 279.06l-71.77-72.31c-7.55-7.6-20.52-2.26-20.52 8.45v48.88h-96v-132.1c0-12.7-5.17-25-14.17-33.99L285.94 14.1c-9-9-21.2-14.1-33.89-14.1H47.99C21.5.1 0 21.6 0 48.09v415.92C0 490.5 21.5 512 47.99 512h287.94c26.5 0 48.07-21.5 48.07-47.99V360.07h-48v103.94H47.99V48.09h159.97v103.98c0 13.3 10.7 23.99 24 23.99H336v88.01H172c-6.63 0-12 5.37-12 12v24c0 6.63 5.37 12 12 12h308v48.88c0 10.71 12.97 16.05 20.52 8.45l71.77-72.31c4.95-4.99 4.95-13.04 0-18.03zM255.95 128.07V51.99l76.09 76.08h-76.09z"
                fill={
                  Array [
                    0,
                    4281348144,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`The SermonNotes component should render with a title, series, and one speaker 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderRadius": 16,
      "marginHorizontal": 16,
      "marginVertical": 12,
      "shadowColor": "rgba(8, 0, 20, 0.25)",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 6,
    }
  }
>
  <View
    style={
      Object {
        "borderRadius": 16,
        "overflow": "hidden",
      }
    }
  >
    <View
      style={
        Object {
          "paddingHorizontal": 16,
          "paddingVertical": 16,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Black",
            "fontSize": 24,
            "lineHeight": 27.6,
          }
        }
      >
        Sermon Notes
      </Text>
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Medium",
            "fontSize": 14,
            "lineHeight": 20.16,
          }
        }
      >
        A Fun Sunday
      </Text>
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Medium",
            "fontSize": 14,
            "lineHeight": 20.16,
          }
        }
      >
        A Fun Series
         - Week 4
      </Text>
      <Text
        style={
          Object {
            "color": "#303030",
            "fontFamily": "Colfax-Medium",
            "fontSize": 14,
            "lineHeight": 20.16,
          }
        }
      >
        Brad Cooper
      </Text>
      <View
        style={
          Object {
            "paddingHorizontal": 16,
            "paddingVertical": 16,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "flex-end",
          "paddingHorizontal": 16,
          "paddingTop": 0,
          "paddingVertical": 16,
        }
      }
    >
      <View
        accessible={true}
        clickable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            bold={false}
            italic={false}
            style={
              Object {
                "color": "#303030",
                "fontFamily": "Colfax-Regular",
                "fontSize": 16,
                "lineHeight": 26,
                "paddingHorizontal": 16,
              }
            }
          >
            Export
          </Text>
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={24}
            bbWidth={24}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                undefined,
                null,
                Object {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={512}
            vbWidth={576}
            width={24}
          >
            <RNSVGGroup
              fill={
                Array [
                  0,
                  4278190080,
                ]
              }
              fillOpacity={1}
              fillRule={1}
              font={Object {}}
              matrix={
                Array [
                  1,
                  0,
                  0,
                  1,
                  0,
                  0,
                ]
              }
              opacity={1}
              originX={0}
              originY={0}
              propList={Array []}
              rotation={0}
              scaleX={1}
              scaleY={1}
              skewX={0}
              skewY={0}
              stroke={null}
              strokeDasharray={null}
              strokeDashoffset={null}
              strokeLinecap={0}
              strokeLinejoin={0}
              strokeMiterlimit={4}
              strokeOpacity={1}
              strokeWidth={1}
              vectorEffect={0}
              x={0}
              y={0}
            >
              <RNSVGPath
                d="M572.29 279.06l-71.77-72.31c-7.55-7.6-20.52-2.26-20.52 8.45v48.88h-96v-132.1c0-12.7-5.17-25-14.17-33.99L285.94 14.1c-9-9-21.2-14.1-33.89-14.1H47.99C21.5.1 0 21.6 0 48.09v415.92C0 490.5 21.5 512 47.99 512h287.94c26.5 0 48.07-21.5 48.07-47.99V360.07h-48v103.94H47.99V48.09h159.97v103.98c0 13.3 10.7 23.99 24 23.99H336v88.01H172c-6.63 0-12 5.37-12 12v24c0 6.63 5.37 12 12 12h308v48.88c0 10.71 12.97 16.05 20.52 8.45l71.77-72.31c4.95-4.99 4.95-13.04 0-18.03zM255.95 128.07V51.99l76.09 76.08h-76.09z"
                fill={
                  Array [
                    0,
                    4281348144,
                  ]
                }
                fillOpacity={1}
                fillRule={1}
                matrix={
                  Array [
                    1,
                    0,
                    0,
                    1,
                    0,
                    0,
                  ]
                }
                opacity={1}
                originX={0}
                originY={0}
                propList={
                  Array [
                    "fill",
                  ]
                }
                rotation={0}
                scaleX={1}
                scaleY={1}
                skewX={0}
                skewY={0}
                stroke={null}
                strokeDasharray={null}
                strokeDashoffset={null}
                strokeLinecap={0}
                strokeLinejoin={0}
                strokeMiterlimit={4}
                strokeOpacity={1}
                strokeWidth={1}
                vectorEffect={0}
                x={0}
                y={0}
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
    </View>
  </View>
</View>
`;
